### ISSUE TITLE
Error compiling package with Intel compiler

### ISSUE THREAD

**Issue by CreRecombinase** [zhengxwen/SNPRelate#27]
Hi, when I try to install this package using the intel compiler I get the following error(s):
```
genBeta.cpp(91): error: expression must have integral or enum type
  					__m256i mask = (g1_1 | ~g1_2) & (g2_1 | ~g2_2);
  					                        ^

genBeta.cpp(91): error: expression must have integral or enum type
  					__m256i mask = (g1_1 | ~g1_2) & (g2_1 | ~g2_2);
  					                                         ^

genBeta.cpp(93): error: expression must have integral or enum type
  					__m256i ibs2 = ~(het | (g1_1 ^ g2_1));
  					                ^

genBeta.cpp(113): error: expression must have integral or enum type
  					__m128i mask = (g1_1 | ~g1_2) & (g2_1 | ~g2_2);
  					                        ^

genBeta.cpp(113): error: expression must have integral or enum type
  					__m128i mask = (g1_1 | ~g1_2) & (g2_1 | ~g2_2);
  					                                         ^

genBeta.cpp(115): error: expression must have integral or enum type
  					__m128i ibs2 = ~(het | (g1_1 ^ g2_1));
```

**Comment by zhengxwen** [comment]
GNU gcc and clang allow bitwise operators with SSE/AVX register variables, but unfortunately intel compiler does not.
I have used these programming grammars a lot in SNPRelate, and there is no quick solution (rewriting the kernal code requires comprehensive testings).

A quick fix may be that you manually "disable" SSE codes in the SNPRelate package:
replace the line of `PKG_CPPFLAGS` in `src/Makevars` and `src/Makevars.win` by:
```
PKG_CPPFLAGS = -DUSING_R -DCOREARRAY_NO_SIMD -I.
```

**Comment by zhengxwen** [comment]
I plan to fix the compiling issues with Intel compilers using SSE2/AXV2 before the next BioC release in Oct.

Best wishes,

**Comment by zhengxwen** [comment]
The compiling errors are fixed in the commit 10b2d5e13353f0ec1ba997b54e838d1c18dd8759.
Let me know if there is still any problem.
